ground/src/test/java/com/google/android/ground/ui/datacollection/tasks/BaseTaskFragmentTest.kt

```diff
@@ -95,6 +96,8 @@ abstract class BaseTaskFragmentTest<F : AbstractTaskFragment<VM>, VM : AbstractT
       preTransactionAction = {
         fragment = this as F
         fragment.taskId = task.id
+        // TODO: Attach `fragment` to mock DataCollectionFragment
```

Comment:
@shobhitagarwal1612 @anandwana001 `DrawAreaTaskFragmentTest` is failing with:

```
Fragment DrawAreaTaskMapFragment{44fdc5b3} (07ec492d-1efe-474f-94ab-b361fdc1063e id=0x2c93 tag=DrawAreaTaskMapFragment) is not a child Fragment, it is directly attached to dagger.hilt.android.internal.managers.ViewComponentManager$FragmentContextWrapper@33500213
java.lang.IllegalStateException: Fragment DrawAreaTaskMapFragment{44fdc5b3} (07ec492d-1efe-474f-94ab-b361fdc1063e id=0x2c93 tag=DrawAreaTaskMapFragment) is not a child Fragment, it is directly attached to dagger.hilt.android.internal.managers.ViewComponentManager$FragmentContextWrapper@33500213
	at androidx.fragment.app.Fragment.requireParentFragment(Fragment.java:1183)
...
```

I believe we have to update the way we mock out the task view model here, but not sure how. Any ideas?

Reply:
Resharing @anandwana001 's suggestion via chat:

adding childfragment:

```kt
inline fun <reified T : Fragment> launchFragmentWithNavController(
  fragmentArgs: Bundle? = null,
  @StyleRes themeResId: Int = R.style.FragmentScenarioEmptyFragmentActivityTheme,
  destId: Int,
  fragment: Fragment?=null,
  crossinline preTransactionAction: Fragment.() -> Unit = {},
  crossinline postTransactionAction: Fragment.() -> Unit = {},
): ActivityScenario<HiltTestActivity> =
  hiltActivityScenario(themeResId).launchFragment<T>(
    fragmentArgs,
    {
      // Attach `fragment` as a child of the mock `DataCollectionFragment`
      parentFragment?.childFragmentManager?.beginTransaction()
        ?.add(R.id.data_collection_fragment, fragment!!, "childFragmentTag")
        ?.commitNow()

      this.preTransactionAction()
      viewLifecycleOwnerLiveData.observeForever { viewLifecycleOwner ->
        if (viewLifecycleOwner != null) {
          val navController = TestNavHostController(getApplicationContext())
          navController.setViewModelStore(ViewModelStore())
          // Required for graph scoped view models.
          navController.setGraph(R.navigation.nav_graph)
          navController.setCurrentDestination(destId, fragmentArgs ?: bundleOf())

          // Bind the controller after the view is created but before onViewCreated is called.
          Navigation.setViewNavController(requireView(), navController)
        }
      }
    },
  ) {
    this.postTransactionAction()
  }
```

and pass the parent fragment

```kt
launchFragmentWithNavController<T>(
  destId = R.id.data_collection_fragment,
  fragment = DataCollectionFragment(),
  preTransactionAction = {
    // Set up fragment-specific properties
    fragment = this as F
    fragment.taskId = [task.id](http://task.id/)
  }
)
```

Reply:
@shobhitagarwal1612 Any ideas on the best way to create and attach `DataCollectionFragment` as the parent to the task under test here? We need to get the view model used by the tasks.

Reply:
@shobhitagarwal1612 Gentle ping